internal/datanode/flush_manager.go

```diff
@@ -96,18 +96,20 @@ var _ flushManager = (*rendezvousFlushManager)(nil)
 type orderFlushQueue struct {
 	sync.Once
 	segmentID UniqueID
-	injectCh  chan *taskInjection
 
 	// MsgID => flushTask
 	working    *typeutil.ConcurrentMap[string, *flushTaskRunner]
 	notifyFunc notifyMetaFunc
 
-	tailMut sync.Mutex
-	tailCh  chan struct{}
-
+	// protect postInjection
 	injectMut     sync.Mutex
-	runningTasks  int32
+	injectCh      chan *taskInjection
 	postInjection postInjectionFunc
+
+	// protect task create and close
+	taskMut      sync.Mutex
+	runningTasks int32
+	tailCh       chan struct{}
```

Comment:
tailCh need to be threadsafe?
waitForAllFlushQueue seems not holding taskMut lock